galasa-ui/src/components/test-runs/SearchCriteriaContent.tsx

```diff
@@ -0,0 +1,269 @@
+    }else {
+      newQuery.set(selectedFilter.id, currentInputValue.trim());
+    }
+
+    // Clean up empty values from the query
+    newQuery.forEach((value, key) => {
+      if (!value) {
+        newQuery.delete(key);
+      }
+    });
+        
+    setQuery(newQuery);
+
+    // Update the URL with the new query parameters
+    const params = new URLSearchParams();
+    newQuery.forEach((value, key) => {
+      params.set(key, value);
+    });
+    router.replace(`${pathname}?${params.toString()}`, {scroll: false});
+  };
+
+
+  const handleCancel = () => {
+    // Revert changes for the currently selected filter
+    if (selectedFilter.id === 'result') {
+      const resultsFromQuery = query.get('result');
+      setSelectedResults(resultsFromQuery ? resultsFromQuery.split(',') : []);
+    } else if (selectedFilter.id === 'status') {
+      const statusesFromQuery = query.get('status');
+      setSelectedStatuses(statusesFromQuery ? statusesFromQuery.split(',') : []);
+    } else if (selectedFilter.id === 'tags') {
+      const tagsFromQuery = query.get('tags');
+      setSelectedTags(tagsFromQuery ? tagsFromQuery.split(',') : []);
+    } else {
+      setCurrentInputValue(query.get(selectedFilter.id) || '');
+    }
+  };
+
+
+  // Render the editor component based on the selected filter
+  const renderComponent = (field: FilterableField) => {
+    // Props for the search component
+    const searchProps = {
+      title: field.description,
+      placeholder: field.placeHolder,
+      value: currentInputValue,
+      onChange: (e: React.ChangeEvent<HTMLInputElement>) => setCurrentInputValue(e.target.value),
+      onClear: () => setCurrentInputValue(''),
+      onSubmit: handleSave,
+      onCancel: handleCancel,
+    };
+
+    // Props for the checkbox list component
+    const checkboxProps = {
+      title: field.description,
+      items: (field.id === 'result') ? resultsNames : TEST_RUNS_STATUS,
+      selectedItems: (field.id === 'result') ? selectedResults : selectedStatuses,
+      onChange: (field.id === 'result') ? setSelectedResults : setSelectedStatuses, 
+      onSubmit: handleSave,
+      onCancel: handleCancel,
+    };
+
+    const tagsPops = {
+      title: field.description,
+      tags: selectedTags,
+      onChange: setSelectedTags,
+      onSubmit: handleSave,
+      onCancel: handleCancel,
+    };
+
+    let customComponent;
+    switch (field.id) {
+    case 'requestor':
+      customComponent = <CustomSearchComponent {...searchProps} allRequestors={allRequestors} />;
+      break;
+    case 'result':
+    case 'status':
+      customComponent = <CustomCheckBoxList {...checkboxProps} />;
+      break;
+    case 'tags':
+      customComponent = <CustomTagsComponent {...tagsPops} />;
+      break;
+    default:
+      customComponent = <CustomSearchComponent {...searchProps} />;
+      break;
+    }
+    return customComponent;
+  };
+
+  return (
+    <div>
```

Comment:
Just tried running this locally myself and noticed that it's a bit difficult to read the text in the selected search criteria row. The blue and black text lacks a bit of contrast, so would it be possible to have the text appear in white when a row is selected, like the "save" text in the button?

![Screenshot 2025-06-26 at 11 20 18](https://github.com/user-attachments/assets/553fb0b3-d8c8-468d-adfc-92c54594ca11)
![Screenshot 2025-06-26 at 11 22 51](https://github.com/user-attachments/assets/4be9caba-e37d-4ba2-b996-a4a86a20a4d1)